### ISSUE TITLE
Function pointers shouldn't trigger IDE1005 (Delegate invocation can be simplified)

### ISSUE THREAD

**Issue by SunnieShine** [dotnet/roslyn#50976]
**Version Used**: 
VS 16.9 Preview 3

**Steps to Reproduce**:

Define a function pointer as a member:

```csharp
public delegate* managed<in Instance, void> FunctionPtr { get; init; }
```

Then use this:

```csharp
if (FunctionPtr != null)
{
    FunctionPtr(this);
}
```

**Expected Behavior**:

Because of function pointer, IDE1005 shouldn't report on this scenario.

**Actual Behavior**:
IDE1005 raised, and suggest us to change the code to `FunctionPtr?.Invoke(this)`.

![1](https://user-images.githubusercontent.com/23616315/106715032-9725bf80-6637-11eb-8c9c-6ee1c667b83a.png)

![2](https://user-images.githubusercontent.com/23616315/106715064-9f7dfa80-6637-11eb-9147-d15a9f46b6c3.png)
